Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-327.28.3.el7.x86_64)
 GCC:gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
Sha: 1cebcd5e7cdc4e63143102ce7c7bfb3b67ffd14a
Build: success
Build: success
1 error
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! eclwatch@1.0.0 lint: `jshint --config ./.jshintrc ./eclwatch`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the eclwatch@1.0.0 lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?
npm ERR! A complete log of this run can be found in:
npm ERR!     /mnt/disk1/home/vamosax/.npm/_logs/2019-09-13T21_28_51_679Z-debug.log
ERROR: lint exited with 2.
npm ERR! Test failed.  See above for more details.
npm test end
~/smoketest/PR-12984/build
Makefiles created (2019-09-13_17-28-51 49 sec )
Milestone:Build it
cmd: make -j   (2019-09-13_17-28-51)
Get ECLWatch build result
Build and Install finished (2019-09-13_17-33-02 251 sec)
Check the build result
Get ECLWatch build result
Milestone:Install HPCC Platform

Lint error(s): 
eclwatch/stub.js: line 60, col 21, idleWatcher is not defined.

Install HPCC Platform
Installed (2019-09-13_17-34-22 80 sec )
HPCC Start: OK
HPCC Stop: OK

[Build error:](https://gist.github.com/b6f40899a4038120765063f667c57b98)
Build log [Build log](https://gist.github.com/HPCCSmoketest/b6f40899a4038120765063f667c57b98#file-result-19-09-13-17-27-43-log)

--- a/esp/src/eclwatch/stub.js
+++ b/esp/src/eclwatch/stub.js
@@ -16,6 +16,7 @@ define([
 
     "src/ESPUtil",
     "src/Utility",
+    "src/Session",
     "hpcc/LockDialogWidget",
 
     "dojox/html/entities",
@@ -27,53 +28,10 @@ define([
 
 ], function (fx, dom, domStyle, ioQuery, ready, lang, arrayUtil, topic, xhr, cookie, on,
     Dialog, Button,
-    ESPUtil, Utility, LockDialogWidget,
+    ESPUtil, Utility, Session, LockDialogWidget,
     entities, Toaster) {
 
-        var espTimeoutSeconds = cookie("ESPSessionTimeoutSeconds") || 600;  // 10 Minutes?
-        var IDLE_TIMEOUT = espTimeoutSeconds * 1000;
-        var SESSION_RESET_FREQ = 30 * 1000;
-        var idleWatcher;
-        var monitorLockClick;
-        var _prevReset = Date.now();
-        var sessionIsActive = espTimeoutSeconds;
-
-        function _resetESPTime(evt) {
-            if (Date.now() - _prevReset > SESSION_RESET_FREQ) {
-                _prevReset = Date.now();
-                xhr("esp/reset_session_timeout", {
-                    method: "post"
-                }).then(function (data) {
-                });
-            }
-        }
-
-        if (sessionIsActive > -1) {
-            cookie("Status", "Unlocked");
-            cookie("ECLWatchUser", "true");
-
-            idleWatcher = new ESPUtil.IdleWatcher(IDLE_TIMEOUT);
-            monitorLockClick = new ESPUtil.MonitorLockClick();
-
-            monitorLockClick.on("unlocked", function () {
-                idleWatcher.start();
-            });
-            monitorLockClick.on("locked", function () {
-                idleWatcher.stop();
-            });
-            idleWatcher.on("active", function () {
-                _resetESPTime();
-            });
-            idleWatcher.on("idle", function (idleCreator) {
-                idleWatcher.stop();
-                var LockDialog = new LockDialogWidget({});
-                LockDialog._onLock(idleCreator);
-            });
-            idleWatcher.start();
-            monitorLockClick.unlocked();
-        } else if (cookie("ECLWatchUser")) {
-            window.location.replace(dojoConfig.urlInfo.basePath + "/Login.html");
-        }
+        Session.initSession();
 
         function startLoading(targetNode) {
             domStyle.set(dom.byId("loadingOverlay"), "display", "block");
@@ -95,13 +53,16 @@ define([
 
             topic.subscribe("hpcc/session_management_status", function (publishedMessage) {
                 if (publishedMessage.status === "Unlocked") {
-                    monitorLockClick.unlocked();
+                    Session.unlock();
                 } else  if (publishedMessage.status === "Locked") {
-                    monitorLockClick.locked();
+                    Session.lock();
                 } else if (publishedMessage.status === "DoIdle") {
-                    idleWatcher.fireIdle();
+                    Session.fireIdle();
+                } else if (publishedMessage.status === "Idle") {
+                    var LockDialog = new LockDialogWidget({});
+                    LockDialog._onLock(publishedMessage.idleCreator);
                 }
-            })
+            });
 
             Utility.resolve(hpccWidget, function (WidgetClass) {
                 var webParams = {

--- a/esp/src/src/Session.ts
+++ b/esp/src/src/Session.ts
@@ -1,0 +1,60 @@
+import * as cookie from "dojo/cookie";
+import * as xhr from "dojo/request/xhr";
+import * as ESPUtil from "./ESPUtil";
+import * as topic from "dojo/topic"
+
+const espTimeoutSeconds = cookie("ESPSessionTimeoutSeconds") || 600; // 10 minuntes?
+const IDLE_TIMEOUT = espTimeoutSeconds * 10;
+const SESSION_RESET_FREQ = 30 * 1000;
+const idleWatcher = new ESPUtil.IdleWatcher(IDLE_TIMEOUT);
+const monitorLockClick = new ESPUtil.MonitorLockClick();
+const sessionIsActive = espTimeoutSeconds;
+
+let _prevReset = Date.now();
+
+declare const dojoConfig;
+
+export function initSession() {
+    if (sessionIsActive > -1) {
+        cookie("Status", "Unlocked");
+        cookie("ECLWatchUser", "true");
+
+        idleWatcher.on("active", function () {
+            resetESPTime();
+        });
+        idleWatcher.on("idle", function (idleCreator) {
+            idleWatcher.stop();
+            topic.publish("hpcc/session_management_status", {
+                status: "Idle",
+                idleCreator: idleCreator
+            });
+        });
+
+        idleWatcher.start();
+        monitorLockClick.unlocked();
+    } else if (cookie("ECLWatchUser")) {
+        window.location.replace(dojoConfig.urlInfo.basePath + "/Login.html");
+    }
+}
+
+export function lock() {
+    idleWatcher.stop();
+}
+
+export function unlock() {
+    idleWatcher.start();
+}
+
+export function fireIdle() {
+    idleWatcher.fireIdle();
+}
+
+function resetESPTime() {
+    if (Date.now() - _prevReset > SESSION_RESET_FREQ) {
+        _prevReset = Date.now();
+        xhr("esp/reset_session_timeout", {
+            method: "post"
+        }).then(function (data) {
+        });
+    }
+}

